hive-event-listeners/apiary-gluesync-listener/src/main/java/com/expediagroup/apiary/extensions/gluesync/listener/service/GlueTableService.java

```diff
@@ -0,0 +1,122 @@
+  private final HiveToGlueTransformer transformer;
+
+  public GlueTableService(AWSGlue glueClient, String gluePrefix) {
+    this.glueClient = glueClient;
+    this.transformer = new HiveToGlueTransformer(gluePrefix);
+    log.debug("ApiaryGlueSync created");
+  }
+
+  public void create(Table table) {
+    CreateTableRequest createTableRequest = new CreateTableRequest()
+        .withTableInput(transformer.transformTable(table))
+        .withDatabaseName(transformer.glueDbName(table));
+    glueClient.createTable(createTableRequest);
+    log.info(table + " table created in glue catalog");
+  }
+
+  public void update(Table table) {
+    boolean skipArchive = shouldSkipArchive(table);
+
+    UpdateTableRequest updateTableRequest = new UpdateTableRequest()
+        .withSkipArchive(skipArchive)
+        .withTableInput(transformer.transformTable(table))
+        .withDatabaseName(transformer.glueDbName(table));
+    glueClient.updateTable(updateTableRequest);
+    log.info(table + " table updated in glue catalog");
+  }
+
+  public void delete(Table table) {
+    DeleteTableRequest deleteTableRequest = new DeleteTableRequest()
+        .withName(table.getTableName())
+        .withDatabaseName(transformer.glueDbName(table));
+    glueClient.deleteTable(deleteTableRequest);
+    log.info(table + " table deleted from glue catalog");
+  }
+
+  /**
+   * Function to copy partitions using BatchCreatePartitionRequest but knowing the partition input list limit is 100,
+   * however input parameter can have larger list
+    */
+  public void copyPartitions(Table table, List<Partition> partitions) {
+    List<PartitionInput> partitionInputs = partitions.stream()
+        .map(this::convertToPartitionInput)
+        .collect(Collectors.toList());
+    int partitionCount = partitionInputs.size();
+    int batchSize = 100;
+    for (int i = 0; i < partitionCount; i += batchSize) {
+      int end = Math.min(i + batchSize, partitionCount);
+      List<PartitionInput> batch = partitionInputs.subList(i, end);
+      BatchCreatePartitionRequest batchCreatePartitionRequest = new BatchCreatePartitionRequest()
+          .withDatabaseName(transformer.glueDbName(table))
+          .withTableName(table.getTableName())
+          .withPartitionInputList(batch);
+      glueClient.batchCreatePartition(batchCreatePartitionRequest);
+    }
+  }
+
+  /**
+   * Function to return all partitions from a table, considering it could reach max results from request and
+   * then convert that into PartitionInput
+   */
+  public List<com.amazonaws.services.glue.model.Partition> getPartitions(Table table) {
+    List<com.amazonaws.services.glue.model.Partition> partitions = new ArrayList<>();
+    String nextToken = null;
+    do {
+      GetPartitionsRequest getPartitionsRequest = new GetPartitionsRequest()
+          .withDatabaseName(transformer.glueDbName(table))
+          .withTableName(table.getTableName())
+          .withMaxResults(1000)
```

Comment:
Why 1000?

Reply:
Because it is the max supported by Glue

Reply:
Then please make the magic number a constant to understand why 1000 and what it is.